primitives/runtime/src/transaction_validity.rs

```diff
@@ -301,4 +409,26 @@ mod tests {
 		// decode back
 		assert_eq!(TransactionValidity::decode(&mut &*encoded), Ok(v));
 	}
+
+	#[test]
+	fn builder_should_prefix_the_tags() {
+		const PREFIX: &str = "test";
+		let a: ValidTransaction = ValidTransaction::for_pallet(PREFIX)
+			.and_requires(1)
+			.and_requires(2)
+			.and_provides(3)
+			.and_provides(4)
+			.propagate(false)
+			.longevity(5)
+			.priority(3)
+			.priority(6)
+			.into();
+		assert_eq!(a, ValidTransaction {
+			propagate: false,
+			longevity: 5,
+			priority: 6,
+			requires: vec![(PREFIX, 1).encode(), (PREFIX, 2).encode()],
```

Comment:
okay, I see that you expect this in the tests as well. Maybe my understanding of prefix was wrong. I thought prefix is added only once.